modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/framework/TestClusterTest.java

```diff
@@ -67,4 +74,19 @@ public void testSimpleQuery() {
             assertNotNull(row);
         }
     }
+
+    /**
+     * Runs a SELECT query with condition.
+     */
+    @Test
+    public void testQueryWithCondition() {
```

Comment:
Not sure I understand why do we need this test

Reply:
The query in the test uses index.

Reply:
I removed the index definition, and this test still passes. Let's make it more robust then